build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -374,6 +374,26 @@ message Action {
   // directory, as well as every subdirectory and content blob referred to, MUST
   // be in the
   // [ContentAddressableStorage][build.bazel.remote.execution.v2.ContentAddressableStorage].
+  //
+  // The build action may assume that all files that are part of the
+  // input root are available for reading (i.e., at least 'r--' or 'r-x'
+  // in terms of UNIX permissions). All directories are available for
+  // writing ('rwx'), meaning that files and directories may be created
+  // and deleted by the build action at any place within the input root.
+  //
+  // In addition to that, the entire input root must be exposed through
+  // a single mount, meaning that calls to rename() on files and
+  // directories within the input root may not fail due to cross-device
+  // links. If the operating system executing the build action supports
+  // hard links, the same applies to link(). Whether directories outside
+  // of the input root (e.g., /tmp) reside on the same file system is
+  // unspecified.
+  //
+  // Due to recent versions of Linux disallowing the creation of hard
+  // links to files without having write access to them (sysctl
+  // fs.protected_hardlinks=1) and input files may not be writable, the
+  // build action may only be permitted to create hard links to files in
```

Comment:
If I'm reading this right, it's a warning about something that may or may not be a constraint? That feels like "too much information" in this context, since there are many other constraints that different execution systems may or may not impose not documented in the API. (Random examples: limits on numbers of inputs, whether or not inputs or input directories may be symlinks, whether or not output symlinks will be "flattened", available disk space).

If there is a minimum requirement here that should be specified (such as "inputs are readable", above), can we pivot the paragraph to that? Otherwise I'd suggest removing it entirely, and documenting it as limitations of a particular implementation rather than API-level limitations.